src/classes/CollisionSolution.ts

```diff
@@ -0,0 +1,203 @@
+import { Battlefield } from './Battlefield';
+import { Bullet } from './Bullet';
+import { CONFIG } from './Config';
+import { Tank } from './Tank';
+
+export class CollisionSolution {
+  public wallList: number[];
+  public tankList: Tank[];
+  public bulletList: Bullet[];
+  public battlefield: Battlefield;
+
+  public constructor(battlefield: Battlefield) {
+    this.wallList = [0, 0, 0, 0];
+    this.tankList = [];
+    this.bulletList = [];
+    this.battlefield = battlefield;
+  }
+
+  public updateBattlefield(battlefield: Battlefield): void {
+    this.battlefield = battlefield;
+    const wall: number[] = [];
+    wall[0] = battlefield.startX - 1;
+    wall[1] = battlefield.finishX + 1;
+    wall[2] = battlefield.startY - 1;
+    wall[3] = battlefield.finishY + 1;
+  }
+
+  public removeBullet(bullet: Bullet): void {
+    this.bulletList = this.bulletList.filter((item) => {
+      return item.id !== bullet.id;
+    });
+  }
+
+  public removeTank(tank: Tank): void {
+    this.tankList = this.tankList.filter((item) => {
+      return item.id !== tank.id;
+    });
+  }
+
+  public hitWallTestBullet(bullet: Bullet): boolean {
+    let hitWallTest: boolean = false;
+    if (bullet.y <= this.wallList[0] || bullet.y >= this.wallList[2] || bullet.x <= this.wallList[3] || bullet.x >= this.wallList[1]) {
+      hitWallTest = true;
+    }
+
+    return hitWallTest;
+  }
+
+  public hitEnemyTestBullet(bullet: Bullet): boolean {
+    let hitEnemyTest: boolean = false;
+    for (const item of this.tankList) {
+      if (item.x === bullet.x && item.y === bullet.y) {
+        if (item.id !== bullet.owner.id) {
+          hitEnemyTest = true;
+          item.health = item.health - bullet.power;
+          if (item.health <= 0) {
+            bullet.owner.killsScore();
```

Comment:
what does it mean? what this method must do? may be you need to think about better name

Reply:
ok. changed to addKillsScore()